lib/modules/decoder/PayloadService.ts

```diff
@@ -57,23 +59,36 @@ export class PayloadService {
 
     const uuid = request.input.args.uuid || uuidv4();
     let valid = true;
+    let state: StateDecoder = StateDecoder.VALID;
+    let errorReason;
 
     try {
       valid = await decoder.validate(payload, request);
 
+      // TODO: Temporary workaround to prevent breaking anything; in the future,
+      // consider modifying the return value of the 'validate' function to return an object
       if (!valid) {
-        return { valid };
+        throw new SkipError("Skip by user defined validation");
       }
     } catch (error) {
       valid = false;
+      errorReason =
+        error.message || (error.toJSON ? error.toJSON().message : undefined);
```

Comment:
Why using `toJSON` ? If it's an Error or a KuzzleErro then we always have the `message` property